Remove label from ServerConnection commands, since all methods take a single composite param struct

koliyo opened the pull request:
Makes the client usage a bit cleaner imo. But is a breaking API change.

mattmassicotte commented:
You know, I have noticed this too and I agree. Label probably never should have been there.

There's actually a way to do this in a less-disruptive way. You add an overload, make the original call the new one, and mark the original as deprecated.

```swift
@available(*, deprecated, renamed: "initialize(_:)")
public func initialize(params: InitializeParams) async throws -> InitializationResponse {
    try await self.initialize(params)
}
```

I'm kind of torn here because that's a lot of work. But I do think its worth it.

koliyo commented:
Fixed!
Got to practice my regex-foo in VSCode

Search:
```
(public func (\w+)\(_ params:((\n|[^{])*))
```

Replace:
```
@available(*, deprecated, renamed: "$2(_:)")
	public func $2(params:$3{
		try await $2(params)
	}

	$1
```

The search expression was a bit complicated by some signatures being more than one line, but I got it working!

mattmassicotte commented:
🫨 that's incredible
